[Traefik Plugin Catalog] Plugin Analyzer has detected a problem.

The plugin was not imported into Traefik Plugin Catalog.

Cause:
```
failed to run the plugin with Yaegi: the load of the plugin takes too much time(10s), or an error, inside the plugin, occurs during the load: 1:21: import "github.com/crazy-goat/go-mesi/servers/traefik" error: /tmp/traefik-plugin-gop1982048009/src/github.com/crazy-goat/go-mesi/servers/traefik/mesi.go:7:2: import "github.com/crazy-goat/go-mesi/mesi" error: /tmp/traefik-plugin-gop1982048009/src/github.com/crazy-goat/go-mesi/mesi/parser.go:41:15: undefined: max
```
Traefik Plugin Analyzer will restart when you will close this issue.

If you believe there is a problem with the Analyzer or this issue is the result of a false positive, please fill an issue on [piceus](https://github.com/traefik/piceus) repository.

--- a/mesi/parser.go
+++ b/mesi/parser.go
@@ -38,13 +38,21 @@ func (c EsiParserConfig) ParseOnly() bool {
 }
 
 func (c EsiParserConfig) DecreaseMaxDepth() EsiParserConfig {
-	c.MaxDepth = max(c.MaxDepth-1, 0)
+	if c.MaxDepth > 0 {
+		c.MaxDepth = c.MaxDepth - 1
+	} else {
+		c.MaxDepth = 0
+	}
 
 	return c
 }
 
 func (c EsiParserConfig) WithElapsedTime(t time.Duration) EsiParserConfig {
-	c.Timeout = max(c.Timeout-t, 0)
+	if c.Timeout-t > 0 {
+		c.Timeout = c.Timeout - t
+	} else {
+		c.Timeout = 0
+	}
 
 	return c
 }
@@ -53,14 +61,20 @@ func (c EsiParserConfig) OverrideConfig(token esiIncludeToken) EsiParserConfig {
 	if token.Timeout != "" {
 		tokenTimeout, err := strconv.ParseFloat(token.Timeout, 64)
 		if err == nil && tokenTimeout > 0 {
-			c.Timeout = min(c.Timeout, time.Duration(tokenTimeout*float64(time.Second)))
+			timeoutLimit := time.Duration(tokenTimeout * float64(time.Second))
+			if c.Timeout > timeoutLimit {
+				c.Timeout = timeoutLimit
+			}
 		}
 	}
 
 	if token.MaxDepth != "" {
 		tokenMaxDepth, err := strconv.Atoi(token.MaxDepth)
 		if err == nil && tokenMaxDepth >= 0 {
-			c.MaxDepth = min(c.MaxDepth, uint(tokenMaxDepth)+1)
+			limit := uint(tokenMaxDepth) + 1
+			if c.MaxDepth > limit {
+				c.MaxDepth = limit
+			}
 		}
 	}
 

--- a/servers/traefik/mesi.go
+++ b/servers/traefik/mesi.go
@@ -74,8 +74,11 @@ func (p *ResponsePlugin) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 
 		// Write the processed response
 		rw.Write([]byte(processedResponse))
+
+		return
 	}
 
+	rw.Write(customWriter.body.Bytes())
 }
 
 type responseWriter struct {

